lte/gateway/c/session_manager/ChargingGrant.cpp

```diff
@@ -237,6 +258,19 @@ ServiceActionType ChargingGrant::final_action_to_action(
   }
 }
 
+ServiceActionType ChargingGrant::final_action_to_action_on_suspension(
+    const ChargingCredit_FinalAction action) const {
+  switch (action) {
+    case ChargingCredit_FinalAction_REDIRECT:
+      return REDIRECT;
+    case ChargingCredit_FinalAction_RESTRICT_ACCESS:
+      return RESTRICT_ACCESS;
+    case ChargingCredit_FinalAction_TERMINATE:
```

Comment:
Why does the FUA action terminate turn into the CONTINUE SERVICE service action type here?

Reply:
That is the whole point of this feature: if we have transient error we do not terminate the service.
But if we have a FUA installed we trigger it.

This is not 3gpp procedure, but per the description of Freddomfi this feature makes sense with the refill procedure when transient errors are used to indicate the user does not have credit

Reply:
If it is a transient error and there is a FUA, why don't we simply follow the FUA action though? It seems like we follow the action for REDIRECT/RESTRICT but not TERMINATE? Isn't that unexpected?